[Bug] Linesearch hidden in scipy 1.14

## Describe the bug
<!-- A clear and concise description of what the bug is. -->

Testing with most recent scipy leads to the following error

```
Run python -m pytest --durations=20 -v test/ ot/ --color=yes --cov=./ --cov-report=xml
ImportError while loading conftest '/home/runner/work/POT/POT/test/conftest.py'.
test/conftest.py:[11](https://github.com/PythonOT/POT/actions/runs/9657485158/job/26636841070?pr=640#step:6:12): in <module>
    from ot.backend import get_backend_list, jax, tf
ot/__init__.py:23: in <module>
    from . import optim
ot/optim.py:22: in <module>
    from scipy.optimize.linesearch import scalar_search_armijo
E   ImportError: cannot import name 'scalar_search_armijo' from 'scipy.optimize.linesearch' (/opt/hostedtoolcache/Python/3.10.[14](https://github.com/PythonOT/POT/actions/runs/9657485158/job/26636841070?pr=640#step:6:15)/x64/lib/python3.10/site-packages/scipy/optimize/linesearch.py)
Error: Process completed with exit code 4.
```

One needs to import now from teh hiden submodule scipy.optimize._linesearch``

### To Reproduce
Steps to reproduce the behavior:
1. ...
2. 

<!-- If you have error messages or stack traces, please provide it here as well -->

#### Screenshots
<!-- If applicable, add screenshots to help explain your problem. -->

#### Code sample
<!-- Ideally attach a minimal code sample to reproduce the decried issue. 
Minimal means having the shortest code but still preserving the bug. -->

### Expected behavior
<!-- A clear and concise description of what you expected to happen. -->


### Environment (please complete the following information):
- OS (e.g. MacOS, Windows, Linux):
- Python version:
- How was POT installed (source, `pip`, `conda`):
- Build command you used (if compiling from source):
- Only for GPU related bugs:
    - CUDA version:
    - GPU models and configuration:
    - Any other relevant information:

Output of the following code snippet:
```python
import platform; print(platform.platform())
import sys; print("Python", sys.version)
import numpy; print("NumPy", numpy.__version__)
import scipy; print("SciPy", scipy.__version__)
import ot; print("POT", ot.__version__)
```

### Additional context
<!-- Add any other context about the problem here. -->

--- a/RELEASES.md
+++ b/RELEASES.md
@@ -22,6 +22,7 @@
 - Fix (F)GW barycenter functions to support computing barycenter on 1 input + deprecate structures as lists (PR #628)
 - Fix line-search in partial GW and change default init to the interior of partial transport plans (PR #602)
 - Fix `ot.da.sinkhorn_lpl1_mm` compatibility with JAX (PR #592)
+- Fiw linesearch import error on Scipy 1.14 (PR #642, Issue #641)
 
 ## 0.9.3
 *January 2024*

--- a/ot/optim.py
+++ b/ot/optim.py
@@ -19,7 +19,7 @@ with warnings.catch_warnings():
     try:
         from scipy.optimize import scalar_search_armijo
     except ImportError:
-        from scipy.optimize.linesearch import scalar_search_armijo
+        from scipy.optimize._linesearch import scalar_search_armijo
 
 # The corresponding scipy function does not work for matrices
 

